x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/view/InMemoryViewService.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.esql.view;
+
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.xpack.esql.expression.function.EsqlFunctionRegistry;
+
+import java.util.Map;
+import java.util.function.Function;
+
+/**
+ * Simple implementation of {@link ClusterViewService} that keeps the views in memory.
+ * This is useful for testing.
+ */
+public class InMemoryViewService extends ViewService {
```

Comment:
This could be in test sources.

NIT:
It would be nice to avoid abstract part and only have cluster state based implementation in production code for simplicity with in-memory version in test overriding specific lookup parts.